src/tools/run-make-support/src/rustdoc.rs

```diff
@@ -0,0 +1,97 @@
+use std::env;
+use std::path::Path;
+use std::process::{Command, Output};
+
+use crate::{add_host_rpath_env, handle_failed_output};
+
+pub fn rustdoc() -> RustdocInvocationBuilder {
+    RustdocInvocationBuilder::new()
+}
+
+#[derive(Debug)]
+pub struct RustdocInvocationBuilder {
```

Comment:
Can you simplify this to `Rustdoc`? I have another PR #122460 which renames `RustcInvocationBuilder` -> `Rustc` (there's no need to have verbose names like these).